pkg/schedule/schedulers/hot_region_solver.go

```diff
@@ -451,10 +451,13 @@ func (bs *balanceSolver) filterHotPeers(storeLoad *statistics.StoreLoadDetail) [
 	return ret
 }
 
-func (bs *balanceSolver) sortHotPeers(firstSort, secondSort []*statistics.HotPeerStat) map[*statistics.HotPeerStat]struct{} {
-	union := make(map[*statistics.HotPeerStat]struct{}, bs.maxPeerNum)
+func sortHotPeers[T any](firstSort, secondSort []*T, maxPeerNum int) map[*T]struct{} {
+	union := make(map[*T]struct{}, maxPeerNum)
 	// At most MaxPeerNum peers, to prevent balanceSolver.solve() too slow.
-	for len(union) < bs.maxPeerNum {
+	for len(union) < maxPeerNum {
+		if len(firstSort) == 0 && len(secondSort) == 0 {
```

Comment:
It‘s better to move this check to the header of this function.

Reply:
<img width="838" alt="image" src="https://github.com/user-attachments/assets/40424dc0-da4d-48aa-ab5e-645a9f0dbcf3" />

They are changing